Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -3793,6 +3792,32 @@ ExceptionOr<void> WebGLRenderingContextBase::texImageSource(TexImageFunctionID f
 }
 #endif
 
+#if ENABLE(OFFSCREEN_CANVAS)
+ExceptionOr<void> WebGLRenderingContextBase::texImageSource(TexImageFunctionID functionID, GCGLenum target, GCGLint level, GCGLint internalformat, GCGLint border, GCGLenum format, GCGLenum type, GCGLint xoffset, GCGLint yoffset, GCGLint zoffset, const IntRect& inputSourceImageRect, GCGLsizei depth, GCGLint unpackImageHeight, OffscreenCanvas& source)
```

Comment:
Is there any reason we can't have a single codepath for `CanvasBase` instead of separate ones for `HTMLCanvasElement` and `OffscreenCanvas`?

Reply:
Yeah, there's no common method to access the pixels.
There should be, but that could be for the next patch.

Reply:
`buffer` and `copiedImage` are both on `CanvasBase`, what else would we need?